File: backend/src/main/java/es/udc/tfgproject/backend/model/services/ProductService.java
package es.udc.tfgproject.backend.model.services;

import es.udc.tfgproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.tfgproject.backend.model.entities.Product;

import java.util.List;

public interface ProductService {

    Product findProductById(Long id) throws InstanceNotFoundException;
    List<Product> findProducts(String keywords, String productState);
    //Producto duplicado?
    Product createProduct(String code, byte[] image, String origin, Float price, String type, String subtype, String productName, String description, String location, String observations);
    Product updateProduct(Long productId, String code, byte[] image, String origin, Float price, String type, String subtype, String productName, String description, String location, String observations) throws InstanceNotFoundException;
    void deleteProduct(Long productId) throws InstanceNotFoundException;
}

File: frontend/src/backend/productService.js
import { appFetch, config } from "./appFetch";

export const findProductById = (id, onSuccess, onErrors) => {
    appFetch(`/products/${id}`, config('GET'), onSuccess, onErrors);
};

export const findProducts = (keywords, productState, onSuccess) => {
    let url = "/products";
    if (keywords) {
        url += `?keywords=${keywords}`;
        if (productState !== undefined && productState !== null) {
            url += `&productState=${productState}`;
        }
    } else if (productState) {
        url += `?productState=${productState}`;
    }
    appFetch(url, config('GET'), onSuccess);
};

export const createProduct = (code, image, origin, price, type, subtype, productName, description, location, observations, onSuccess, onErrors) =>
    appFetch('/products', config('POST', {code, image, origin, price, type, subtype, productName, description, location, observations}), onSuccess, onErrors);

export const updateProduct = (id, code, image, origin, price, type, subtype, productName, description, location, observations, onSuccess, onErrors) =>
    appFetch(`/products/${id}`, config('PUT', {code, image, origin, price, type, subtype, productName, description, location, observations}), onSuccess, onErrors);

export const deleteProduct = (id, onSuccess, onErrors) =>
    appFetch(`/products/${id}`, config('DELETE'), onSuccess, onErrors);
